### PR TITLE
Update UG and DG

### PR DESCRIPTION
**Pull request by nseah21**
### Changes made for DG:
1. Added section on colour coding for DG diagrams
2. Updated UML diagram for Model component
3. Updated description for said UML diagram 
4. Removed "more OOP design" section from Model component
5. Added use cases for show/hide/reset
6. Removed `list /i` and `list /e` from `list` (commented out -- to delete after finalising the removal of the features from RC4HDB)
7. Added manual testing guide for list and show/hide/reset

### Changes made for UG:
1. Removed any mentions to `list /i` and `list /e` under viewing residents section
2. Updated FAQ with choice of single letters as inputs for column manipulator commands

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -332,33 +333,43 @@ As a result, any updates to `ObservableList<Resident>` would be reflected immedi
...
-**Challenges faced with UI:**
-
-The original UI represented a `Person` field as a nested FXML `Label` within a `PersonCard`, which was used to populate the `ListView` panel. Calling `setVisible` on a `Label` resulted in blank gaps in the panel because the `Label` was ultimately still intact, just *invisible*. Hence, there was a need to find another method to hide and collapse rows/columns.
+**Changes to UI component:**
 
-**Gaps in** `ListView` **panel:**
+To achieve the show/hide feature for resident fields, we first had to rework our UI to use a `TableView`
+and two `ObservableList<String>` objects, instead of a `ListView` and a `PersonCard`. This was because the `setVisible`
+method does not work properly with a `ListView` and a `PersonCard` -- as shown below, hidden fields would be rendered 
+as gaps in the `ListView`, rather than being omitted entirely.
```
Comment (nealetham) [inline docs/DeveloperGuide.md:341]: I thought we should not mention any past implementations, i.e. `ListView` and `PersonCard`

**Review by nealetham — CHANGES_REQUESTED** [review]
Take a look at the comments.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (nseah21) [reply docs/DeveloperGuide.md]: Hi Neale, I've removed this section on the UI implementation, since you have also mentioned it in your PR.

**Review by nealetham — APPROVED** [review]
LGTM, but let's see how it'll look in conjunction with the rest of our UG after you merge it!

**Pull request merged**